### DISCUSSION TITLE
[@types/node] nodejs Error Constructor does not accept more than one argument

### DISCUSSION THREAD

**Discussion by moeenn** [DefinitelyTyped/DefinitelyTyped#60780] [Issues with a @types package]
Reading the MDN, I noticed that we can pass multiple arguments to the `Error` constructor

[MDN - Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#providing_structured_data_as_the_error_cause)

- The first argument is the error message
- The second argument is an object which includes the property `cause`

Testing out whether the Node implementation of `Error` allows this, I came up with this code

```javascript
const cause = { id: 1, example: "other" }
const err = new Error("Misc. Error description", { cause })

console.log(err.message)  // Misc. Error description
console.log(err.cause) // { id: 1, example: "other" }
```

I tried instantiating an `Error` instance as follows 
```typescript
new Error(
  "Required environment variables not set",
  { cause },
)
```

I got the following error
```
Expected 0-1 arguments, but got 2.ts(2554)
```

It appears that the `@types/node` package should be updated to accommodate this.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
I believe both web and Node types are using same constructor definition (no additional arameters):
https://github.com/microsoft/TypeScript/blob/main/lib/lib.es5.d.ts#L979-L980
(I think @types/node just augments default ErrorConstructor interface with additiona symbols, constructor stamp is the same).
If anything, this should be transferred to TypeScript issue, as it touches library definition.

**Reply by mcollina** [reply]
I agree

**Comment by moeenn** [comment]
I see. I guess I'll have to take it up with Microsoft.

**Reply by thernstig** [reply]
@moeenn do you have a link to where you brought this up? https://github.com/microsoft/TypeScript/issues?page=3&q=is%3Aissue+is%3Aopen+error+cause gives so many pages of hits that it is hard to find.

**Reply by NemoStein** [reply]
Couldn't find either, so I opened one:
https://github.com/microsoft/TypeScript/issues/53465
